docs/getting-started/rulesets.md

```diff
@@ -199,3 +198,141 @@ rules:
+If you need to access environment specific API, make sure you provide an alternative for other environments.
+A good example of such situation is `fetch` - a function available natively in a browser context, but missing in Node.js.
+To keep your code cross-platform, you shall consider using a package like node-fetch or isomorphic-fetch, which implement spec-compliant Fetch API and work in Node.js.
+You are obviously allowed to use anything that's available natively in JS (ES2017), this is naturally a safe usage.
+
+
+Besides that, we encourage you to not transpile the code to ES5 if you can help it. Spectral does not support older environments than ES2017, so there is no need to bloat the bundle with useless transformations and polyfills.
+Ship untransformed async/await, do not include unneeded shims, it's all good.
+
+Another caveat is that ES Modules and other modules systems are not supported. Although, you are recommended to write ES2017 code, you should not be using require or imports.
+To give you an example of a good code
+```js
+module.exports = (obj) => {
+  for (const [key, value] of Object.entries(obj)) {
+    // this is a perfectly fine code
+  }
+};
+```
+You do not need to provide any shim for `Object.entries` or use [regenerator](https://facebook.github.io/regenerator/) for `for of` loop.
+As stated, ou cannot use ES Modules, so the following code is considered sa invalid and won't work correctly.
+```js
+export default (obj) => {
+  for (const [key, value] of Object.entries(obj)) {
+    // this is a perfectly fine code
+  }
+};
+```
+The following code won't work properly either
+ ```js
+const foo = require('./foo'); // require is not available (see note below)
+
+module.exports = (obj) => {
+  for (const [key, value] of Object.entries(obj)) {
+    // this is a perfectly fine code
+  }
+};
+```
+
+If you have any module system, you need to use some bundler, preferably Rollup.js as it generates efficient bundles.
+We are still evaluating the idea of supporting ESModule and perhaps we decide to bring support for ES Modules at some point,
+yet for now you cannot use them.
+
+#### How do I create a function?
+
+The process of creating a function involves 2 steps:
+
+- create a js file inside of a directory called `functions` that should be placed next to your ruleset file
+- create a `functions` array in your ruleset if you haven't done it yet and place a string with the filename without `.js` extension
+
+Example:
+
+`functions/abc.js`
+
+```js
+module.exports = (targetVal) => {
+  if (targetVal !== 'abc') {
+    return [
+      {
+        message: 'Value must equal "abc".',
+      },
+    ];
+  }
+};
+```
+
+`my-ruleset.yaml`
+
+```yaml
+functions: [abc]
+rules:
+  my-rule:
+    message: "{{error}}"
+    given: "$.info"
+    then:
+      function: "abc"
+```
+
+Optionally, if you'd like to validate the data that it passed to abc function before the function gets executed, you can provide a JSON Schema.
```

Comment:
This is basically documenting the options that the function accepts, right? Not validating the incoming `targetVal`. Assuming this, suggest re-wording to make it more clear.

Reply:
It validates target.